Dockerfile.dapper

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04
+FROM ubuntu:22.04
```

Comment:
Can we use BCI here?

Reply:
That should not be a problem, I did not do it as part of this PR given the sheer amount of changes it already includes.

Reply:
Using BCI will make sure it is built using the latest Golang automatically versus manual bumps, same for Docker binaries. As long as its followed up, it can be in multiple stages.

Reply:
Moved dapper base image to `bci-base`, PTAL.